unity-renderer/Assets/DCLPlugins/Social/Chat/Channels/ChannelLinkDetector.cs

```diff
@@ -10,34 +9,29 @@
 
 namespace DCL.Social.Chat
 {
-    public class ChannelLinkDetector : MonoBehaviour, IPointerClickHandler
+    public class ChannelLinkDetector : MonoBehaviour, IPointerClickHandler, IChannelLinkDetectorView
     {
+        public static readonly BaseHashSet<ChannelLinkDetector> INSTANCES = new ();
```

Comment:
`ChannelLinkDetector` is part of the prefabs hierarchy, there is not explicit instantiation through code, thus it should remain this way since it needs a `TMP_Text` component to work, which is also part of the prefab hierarchy.
Given this limitation, can you think of a better way of refactoring into MVC? So we can avoid implicit coupling with singletons inside `ChannelLinkDetector`

Reply:
I will check on my own to propose the exact structure but off the top of my head:

- Expose these fields from the root view to the root controller (one that handles the upper-level view)
- Make it instantiate sub-controllers for each exposed field on initialization

Reply:
Im not completely following. Can you provide an example?

Reply:
For example:
You have a root view: `PlaceCardComponentView`.
You already have `PlacesAndEventsCardsFactory` you create this view with.
From the view you can expose `IReadOnlyList<ChannelLinkDetector>` (just by `GetComponentInChildren`)
In the factory or in `PlacesCardsConfigurator` you can access all required dependencies and create `ChannelLinkDetectorController` for each exposed view.

Reply:
I dont find this approach maintainable in the long term. `PlaceCardComponentView` is not the only prefab that uses this component, but also the chat system. For every time `ChannelLinkDetector` which is added into a prefab, we would have to make a factory exposing the components, which could be quite tedious.
I could propose to revert the refactor and keep everything in the view for the sake of simplicity and manteinance.

Reply:
Reverted refactor